torch/lib/c10d/CUDAUtils.hpp

```diff
@@ -16,7 +16,7 @@ class CUDAEvent {
 
   CUDAEvent() : CUDAEvent(nullptr, 0) {}
 
-  ~CUDAEvent();
+  ~CUDAEvent() noexcept(false);
```

Comment:
Please no... Can we make our destructors noexcept? @pietern

Reply:
I agree with @ezyang , but also made a new issue for future discussion on this. https://github.com/pytorch/pytorch/issues/11271

Reply:
@apaszke The alternative is not checking the error at all. Pick your poison. I've been bitten by cascading exceptions from dtors in the past so I'm fine removing the check. That said, not checking the exception at all is not a great outcome either.

Reply:
I'm sure there's a ton of other dtors that are throwing. If we go the noexcept route we should apply it absolutely everywhere, or it will have been for naught.

Reply:
Yeah, I agree with @pietern